src/sorted-table.js

```diff
@@ -0,0 +1,49 @@
+import React from "react";
+import {useState} from "react";
+import "./sorted-table.css";
+
+export default function SortedTable({tableData}) {
+
+    const tableContentWithKey = () => tableData.slice(1).map((row, i) => [i].concat(row));
+    const compare = (a,b) =>
+        typeof a === "number" ? a - b
+            : typeof a === "string" ? (a.toLowerCase() < b.toLowerCase() ? -1 : a.toLowerCase() > b.toLowerCase() ? 1 : 0)
+            : typeof a === "boolean" ? (a && !b ? 1 : a === b ? 0 : -1)
+            : 0;
+
+    const [sortColumn, setSortColumn] = useState(0);
+    const [tableContent, setTableContent] = useState(tableContentWithKey());
+    const [ascending, setAscending] = useState(true);
+
+
+
+    function sortTable(tbl, column, asc) {
+        return tbl.map((row, i) => [row[column], i])
+            .sort((a, b) => compare(a[0], b[0]) * (asc ? 1 : -1))
```

Comment:
I got lost here. I think this is overcomplicated and quite hard to follow the logic.
I would suggest to think of a simpler way to do it.

Reply:
You're right. I modified a bit to make it more readable